In [1]:
import numpy as np
import pandas as pd
import re

In [2]:
pd.options.display.float_format = '{:.2f}'.format

### Companies Data Analysis

In [3]:
df_com = pd.read_csv('companies.txt', sep='\t', encoding='ISO-8859-1')

In [4]:
df_com.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB


In [5]:
df_com.describe()

permalink   name                homepage_url category_list  \
count                  66368  66367                       61310         63220   
unique                 66368  66102                       61191         27296   
top     /Organization/Skidos  Peach  http://www.askforoffer.com      Software   
freq                       1      4                           5          3995   

           status country_code state_code       region           city  \
count       66368        59410      57821        58338          58340   
unique          4          137        311         1092           5111   
top     operating          USA         CA  SF Bay Area  San Francisco   
freq        53034        37601      12900         8804           3526   

        founded_at  
count        51147  
unique        3978  
top     01-01-2012  
freq          2730

In [6]:
df_com.size

663680

In [7]:
df_com.shape

(66368, 10)

In [8]:
df_com.columns

Index(['permalink', 'name', 'homepage_url', 'category_list', 'status',
       'country_code', 'state_code', 'region', 'city', 'founded_at'],
      dtype='object')

In [9]:
df_com.head(10)

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   
5     /Organization/01Games-Technology      01Games Technology   
6  /Organization/0Ndine-Biomedical-Inc  Ondine Biomedical Inc.   
7                 /Organization/0Xdata                  H2O.ai   
8                      /Organization/1                One Inc.   
9            /Organization/1-2-3-Listo             1,2,3 Listo   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   
5             http://www.01games.hk/   
6               http://ondinebio.com   
7                     http://h2o.ai/   
8                 http://whatis1.com   
9            http://www.123listo.com   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   
5                                              Games  operating          HKG   
6                                      Biotechnology  operating          CAN   
7                                          Analytics  operating          USA   
8                                             Mobile  operating          USA   
9                                         E-Commerce  operating          CHL   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010  
5        NaN              Hong Kong      Hong Kong         NaN  
6         BC              Vancouver      Vancouver  01-01-1997  
7         CA            SF Bay Area  Mountain View  01-01-2011  
8         CA            SF Bay Area  San Francisco  01-08-2011  
9         12               Santiago     Las Condes  01-01-2012

If we observe the permalink, the case is not uniform; lets convert into lower case

In [10]:
df_com['permalink'] = df_com['permalink'].str.lower()

In [11]:
df_com.head(10)

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
3                /organization/0-6-com                 0-6.com   
4       /organization/004-technologies        004 Technologies   
5     /organization/01games-technology      01Games Technology   
6  /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   
7                 /organization/0xdata                  H2O.ai   
8                      /organization/1                One Inc.   
9            /organization/1-2-3-listo             1,2,3 Listo   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   
5             http://www.01games.hk/   
6               http://ondinebio.com   
7                     http://h2o.ai/   
8                 http://whatis1.com   
9            http://www.123listo.com   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   
5                                              Games  operating          HKG   
6                                      Biotechnology  operating          CAN   
7                                          Analytics  operating          USA   
8                                             Mobile  operating          USA   
9                                         E-Commerce  operating          CHL   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010  
5        NaN              Hong Kong      Hong Kong         NaN  
6         BC              Vancouver      Vancouver  01-01-1997  
7         CA            SF Bay Area  Mountain View  01-01-2011  
8         CA            SF Bay Area  San Francisco  01-08-2011  
9         12               Santiago     Las Condes  01-01-2012

## As per the description of the Companies data, Permalink is unique to each company

## Answer to the Question - How many unique companies are present in the companies file?

In [12]:
unique_permalink = df_com.permalink.unique()
print(len(unique_permalink))

66368


In [13]:
df_com['name'] = df_com['name'].str.lower()

In [14]:
df_com.head(15)

permalink                         name  \
0                        /organization/-fame                        #fame   
1                     /organization/-qounter                     :qounter   
2        /organization/-the-one-of-them-inc-       (the) one of them,inc.   
3                      /organization/0-6-com                      0-6.com   
4             /organization/004-technologies             004 technologies   
5           /organization/01games-technology           01games technology   
6        /organization/0ndine-biomedical-inc       ondine biomedical inc.   
7                       /organization/0xdata                       h2o.ai   
8                            /organization/1                     one inc.   
9                  /organization/1-2-3-listo                  1,2,3 listo   
10                     /organization/1-4-all                      1-4 all   
11            /organization/1-618-technology             1.618 technology   
12               /organization/1-800-dentist                1-800-dentist   
13               /organization/1-800-doctors                1-800-doctors   
14  /organization/1-800-publicrelations-inc-  1-800-publicrelations, inc.   

                          homepage_url  \
0                   http://livfame.com   
1               http://www.qounter.com   
2                  http://oneofthem.jp   
3                   http://www.0-6.com   
4    http://004gmbh.de/en/004-interact   
5               http://www.01games.hk/   
6                 http://ondinebio.com   
7                       http://h2o.ai/   
8                   http://whatis1.com   
9              http://www.123listo.com   
10                                 NaN   
11  http://www.Homeandcondogallery.com   
12          http://www.1800dentist.com   
13              http://1800doctors.com   
14  http://www.1800publicrelations.com   

                                        category_list     status country_code  \
0                                               Media  operating          IND   
1   Application Platforms|Real Time|Social Network...  operating          USA   
2                                   Apps|Games|Mobile  operating          NaN   
3                                         Curated Web  operating          CHN   
4                                            Software  operating          USA   
5                                               Games  operating          HKG   
6                                       Biotechnology  operating          CAN   
7                                           Analytics  operating          USA   
8                                              Mobile  operating          USA   
9                                          E-Commerce  operating          CHL   
10                       Entertainment|Games|Software  operating          USA   
11                 Networking|Real Estate|Web Hosting  operating          USA   
12                                Health and Wellness  operating          USA   
13                                Health and Wellness  operating          USA   
14          Internet Marketing|Media|Public Relations  operating          USA   

   state_code                 region               city  founded_at  
0          16                 Mumbai             Mumbai         NaN  
1          DE             DE - Other      Delaware City  04-09-2014  
2         NaN                    NaN                NaN         NaN  
3          22                Beijing            Beijing  01-01-2007  
4          IL  Springfield, Illinois          Champaign  01-01-2010  
5         NaN              Hong Kong          Hong Kong         NaN  
6          BC              Vancouver          Vancouver  01-01-1997  
7          CA            SF Bay Area      Mountain View  01-01-2011  
8          CA            SF Bay Area      San Francisco  01-08-2011  
9          12               Santiago         Las Condes  01-01-2012  
10         NC             NC - Other  Connellys Springs       

In [15]:
unique_name = df_com.name.unique()
len(unique_name)

66038

In [16]:
df_dup_com = df_com[df_com.duplicated(['name'])]

In [17]:
df_dup_com.head()

permalink    name               homepage_url  \
282    /organization/3divaz-3  3divaz  http://www.3divaz.ch/Home   
1527   /organization/adtena-2  adtena          http://adtena.com   
1996    /organization/agora-6   agora       https://www.agora.co   
2460    /organization/alike-2   alike           http://alike.fr/   
3006  /organization/amicus-co  amicus       http://www.amicus.co   

                     category_list     status country_code state_code  \
282                            NaN  operating          CHE          1   
1527                           NaN     closed          NaN        NaN   
1996  Android|Apps|Internet|Mobile  operating          USA         NY   
2460                        Design     closed          FRA         A8   
3006   Apps|Mobile|Mobile Commerce  operating          IND          7   

             region       city  founded_at  
282     CHE - Other    Wildegg  01-07-2014  
1527            NaN        NaN  15-11-2014  
1996  New York City   New York  01-07-2013  
2460          Paris      Paris         NaN  
3006      New Delhi  New Delhi  01-01-2015

In [18]:
# df_dup_com1 = df_com[df_com.duplicated()]

In [19]:
# df_dup_com1.shape

In [20]:
df_dup_com.shape

(330, 10)

There are around 330 companies which are duplicated; lets filter that later and lets continue using the company table
with duplicates

In [21]:
df_com.iloc[281]

permalink           /organization/3divaz-2
name                                3divaz
homepage_url     http://www.3divaz.ch/Home
category_list                          NaN
status                              closed
country_code                           NaN
state_code                             NaN
region                                 NaN
city                                   NaN
founded_at                      01-07-2014
Name: 281, dtype: object

In [22]:
df_com.iloc[282]

permalink           /organization/3divaz-3
name                                3divaz
homepage_url     http://www.3divaz.ch/Home
category_list                          NaN
status                           operating
country_code                           CHE
state_code                               1
region                         CHE - Other
city                               Wildegg
founded_at                      01-07-2014
Name: 282, dtype: object

In [23]:
# TODO - display all duplicated companies

## Rounds

In [24]:
df_rounds = pd.read_csv('rounds2.csv', encoding='ISO-8859-1')

In [25]:
df_rounds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB


In [26]:
df_rounds.describe()

raised_amount_usd
count           94959.00
mean         10426869.33
std         114821247.98
min                 0.00
25%            322500.00
50%           1680511.00
75%           7000000.00
max       21271935000.00

In [27]:
df_rounds.shape

(114949, 6)

In [28]:
df_rounds.columns

Index(['company_permalink', 'funding_round_permalink', 'funding_round_type',
       'funding_round_code', 'funded_at', 'raised_amount_usd'],
      dtype='object')

In [29]:
df_rounds.head(10)

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   
5       /ORGANIZATION/004-TECHNOLOGIES   
6     /organization/01games-technology   
7  /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8  /organization/0ndine-biomedical-inc   
9                 /ORGANIZATION/0XDATA   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9  /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015        10000000.00  
1                  A  14-10-2014                nan  
2                NaN  01-03-2014          700000.00  
3                  B  30-01-2014         3406878.00  
4                  A  19-03-2008         2000000.00  
5                NaN  24-07-2014                nan  
6                NaN  01-07-2014           41250.00  
7                NaN  11-09-2009           43360.00  
8                NaN  21-12-2009          719491.00  
9                NaN  22-05-2013         3000000.00

In [30]:
df_rounds['company_permalink'] = df_rounds['company_permalink'].str.lower()

In [31]:
df_rounds.head(10)

company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   
5       /organization/004-technologies   
6     /organization/01games-technology   
7  /organization/0ndine-biomedical-inc   
8  /organization/0ndine-biomedical-inc   
9                 /organization/0xdata   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9  /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015        10000000.00  
1                  A  14-10-2014                nan  
2                NaN  01-03-2014          700000.00  
3                  B  30-01-2014         3406878.00  
4                  A  19-03-2008         2000000.00  
5                NaN  24-07-2014                nan  
6                NaN  01-07-2014           41250.00  
7                NaN  11-09-2009           43360.00  
8                NaN  21-12-2009          719491.00  
9                NaN  22-05-2013         3000000.00

In [32]:
df_rounds.company_permalink.shape

(114949,)

## Answer to the Question - How many unique companies are present in rounds2?

In [33]:
unique_comp_perm = df_rounds.company_permalink.unique()
print(len(unique_comp_perm))

66370


## Analysis - Q.Are there any companies in the rounds2 file which are not  present in companies ? Answer Y/N.

As per the below calculation, there are 2 companies missing in companies data compared to Rounds data.
Also the Missing comapnies listed

In [34]:
set_company_permlink = set(unique_comp_perm)

In [35]:
set_permlink = set(unique_permalink)

In [36]:
len(set_company_permlink)

66370

In [37]:
len(set_permlink)

66368

In [38]:
# 2 companies missing in companies table compared to the rounds table

#### Data present in Rounds table which isnt present in companies table

In [39]:
set_company_permlink.difference(set_permlink)

{'/organization/e-cã\x8abica',
 '/organization/energystone-games-ç\x81µç\x9f³æ¸¸æ\x88\x8f',
 '/organization/huizuche-com-æ\x83\xa0ç§ÿ车',
 '/organization/magnet-tech-ç£\x81ç\x9f³ç§\x91æ\x8a\x80',
 '/organization/tipcat-interactive-æ²\x99è\x88ÿä¿¡æ\x81¯ç§\x91æ\x9a\x80',
 '/organization/weiche-tech-å\x96\x82车ç§\x91æ\x8a\x80',
 '/organization/zengame-ç¦\x85游ç§\x91æ\x8a\x80'}

#### Another Way

In [40]:
set_diff_df = pd.concat([df_rounds['company_permalink'], df_com['permalink'], df_com['permalink']]).drop_duplicates(keep=False)
print(set_diff_df)

29597                               /organization/e-cãŠbica
31863          /organization/energystone-games-灵石游戏
45176                  /organization/huizuche-com-惠ç§ÿ车
58473                /organization/magnet-tech-磁石科技
101036    /organization/tipcat-interactive-沙èˆÿ信息ç...
109969               /organization/weiche-tech-喂车科技
113839                   /organization/zengame-禅游科技
dtype: object


In [41]:
# Even though the diff is 2; the actual difference is 7 (len(set_diff_df))
# so after merge 114949-7 rows should be pesent in the merged df

In [42]:
# data in companies which isnt present in rounds
set_permlink.difference(set_company_permlink)

{'/organization/e-cã\x9abica',
 '/organization/energystone-games-ç\x81µçÿ³æ¸¸æ\x88\x8f',
 '/organization/huizuche-com-æ\x83\xa0ç§\x9f车',
 '/organization/tipcat-interactive-æ²\x99è\x88\x9fä¿¡æ\x81¯ç§\x91æ\x9a\x80',
 '/organization/zengame-ç¦\x85游ç§\x91æ\x9a\x80'}

#### merge the data frames
Need to complete the information of rounds from the company df. So have to join both the df based on the intersection values 

In [43]:
master_frame = pd.merge(df_rounds, df_com, how='inner', left_on='company_permalink', right_on='permalink')

In [44]:
master_frame.head(15)

company_permalink  \
0                   /organization/-fame   
1                /organization/-qounter   
2                /organization/-qounter   
3   /organization/-the-one-of-them-inc-   
4                 /organization/0-6-com   
5        /organization/004-technologies   
6      /organization/01games-technology   
7   /organization/0ndine-biomedical-inc   
8   /organization/0ndine-biomedical-inc   
9                  /organization/0xdata   
10                 /organization/0xdata   
11                 /organization/0xdata   
12                 /organization/0xdata   
13                      /organization/1   
14                      /organization/1   

                            funding_round_permalink funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1   /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2   /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3   /funding-round/650b8f704416801069bb178a1418776b            venture   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5   /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6   /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7   /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9   /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
12  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
13  /funding-round/03b975068632eba5bfdb937ec8c07a68               seed   
14  /funding-round/5de6d8828aef7d925d97918d15727670        undisclosed   

   funding_round_code   funded_at  raised_amount_usd  \
0                   B  05-01-2015        10000000.00   
1                   A  14-10-2014                nan   
2                 NaN  01-03-2014          700000.00   
3                   B  30-01-2014         3406878.00   
4                   A  19-03-2008         2000000.00   
5                 NaN  24-07-2014                nan   
6                 NaN  01-07-2014           41250.00   
7                 NaN  11-09-2009           43360.00   
8                 NaN  21-12-2009          719491.00   
9                 NaN  22-05-2013         3000000.00   
10                  B  09-11-2015        20000000.00   
11                NaN  03-01-2013         1700000.00   
12                  A  19-07-2014         8900000.00   
13                NaN  05-02-2014          150000.00   
14                NaN  05-02-2013                nan   

                              permalink                    name  \
0                   /organization/-fame                   #fame   
1                /organization/-qounter                :qounter   
2                /organization/-qounter                :qounter   
3   /organization/-the-one-of-them-inc-  (the) one of them,inc.   
4                 /organization/0-6-com                 0-6.com   
5        /organization/004-technologies        004 technologies   
6      /organization/01games-technology      01games technology   
7   /organization/0ndine-biomedical-inc  ondine biomedical inc.   
8   /organization/0ndine-biomedical-inc  ondine biomedical inc.   
9                  /organization/0xdata                  h2o.ai   
10                 /organization/0xdata                  h2o.ai   
11                 /organization/0xdata                  h2o.ai   
12                 /organization/0xdata                  h2o.ai   
13                      /organization/1                one inc.   
14                      /organization/1                one inc.   

                         homepage_url  \
0                  http://livfame.com   
1              http://www.qounter.com   
2              http://www.qounter.com   
3             

In [45]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 114942 entries, 0 to 114941
Data columns (total 16 columns):
company_permalink          114942 non-null object
funding_round_permalink    114942 non-null object
funding_round_type         114942 non-null object
funding_round_code         31139 non-null object
funded_at                  114942 non-null object
raised_amount_usd          94958 non-null float64
permalink                  114942 non-null object
name                       114941 non-null object
homepage_url               108810 non-null object
category_list              111535 non-null object
status                     114942 non-null object
country_code               106271 non-null object
state_code                 104003 non-null object
region                     104782 non-null object
city                       104785 non-null object
founded_at                 94422 non-null object
dtypes: float64(1), object(15)
memory usage: 14.9+ MB


### [Answer] Q . Merge the two data frames so that all  variables (columns)  in the companies frame are added to the rounds2 data frame. Name the merged frame master_frame. How many observations are present in master_frame ?

In [46]:
master_frame.shape

(114942, 16)

In [47]:
print(df_rounds.shape)
print(df_rounds.columns)

(114949, 6)
Index(['company_permalink', 'funding_round_permalink', 'funding_round_type',
       'funding_round_code', 'funded_at', 'raised_amount_usd'],
      dtype='object')


In [48]:
print(df_com.shape)
print(df_com.columns)

(66368, 10)
Index(['permalink', 'name', 'homepage_url', 'category_list', 'status',
       'country_code', 'state_code', 'region', 'city', 'founded_at'],
      dtype='object')


In [99]:
master_frame.to_csv('master.csv')

#### DAta clean-up for further analysis

In [50]:
master_frame.isnull().all()

company_permalink          False
funding_round_permalink    False
funding_round_type         False
funding_round_code         False
funded_at                  False
raised_amount_usd          False
permalink                  False
name                       False
homepage_url               False
category_list              False
status                     False
country_code               False
state_code                 False
region                     False
city                       False
founded_at                 False
dtype: bool

In [51]:
master_frame.isnull().any()

company_permalink          False
funding_round_permalink    False
funding_round_type         False
funding_round_code          True
funded_at                  False
raised_amount_usd           True
permalink                  False
name                        True
homepage_url                True
category_list               True
status                     False
country_code                True
state_code                  True
region                      True
city                        True
founded_at                  True
dtype: bool

In [52]:
master_frame.isnull().sum()

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83803
funded_at                      0
raised_amount_usd          19984
permalink                      0
name                           1
homepage_url                6132
category_list               3407
status                         0
country_code                8671
state_code                 10939
region                     10160
city                       10157
founded_at                 20520
dtype: int64

In [53]:
round(100*master_frame.isnull().sum()/len(master_frame.index),2)

company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        72.91
funded_at                  0.00
raised_amount_usd         17.39
permalink                  0.00
name                       0.00
homepage_url               5.33
category_list              2.96
status                     0.00
country_code               7.54
state_code                 9.52
region                     8.84
city                       8.84
founded_at                17.85
dtype: float64

In [58]:
# Not used any where - funding_round_code, founded_at, city, state_Code, region, homepage_url

In [54]:
master_frame.drop('funding_round_code', axis=1, inplace=True)

In [55]:
master_frame.drop('founded_at', axis=1, inplace=True)

In [59]:
master_frame.drop('city', axis=1, inplace=True)

In [60]:
master_frame.drop('state_code', axis=1, inplace=True)

In [61]:
master_frame.drop('region', axis=1, inplace=True)

In [62]:
master_frame.drop('homepage_url', axis=1, inplace=True)

In [63]:
master_frame['permalink'].equals(master_frame['company_permalink'])

True

In [64]:
# one can be removed

In [65]:
master_frame.drop('company_permalink', axis=1, inplace=True)

In [66]:
round(100*master_frame.isnull().sum()/len(master_frame.index),2)

funding_round_permalink    0.00
funding_round_type         0.00
funded_at                  0.00
raised_amount_usd         17.39
permalink                  0.00
name                       0.00
category_list              2.96
status                     0.00
country_code               7.54
dtype: float64

In [67]:
# raised_amt = master_frame.raised_amount_usd
# raised_amt.isnull()

In [68]:
# raised_amt[1]

In [69]:
master_frame.dropna(subset=['raised_amount_usd'], inplace=True)

In [70]:
master_frame.head()

funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   

    funded_at  raised_amount_usd                            permalink  \
0  05-01-2015        10000000.00                  /organization/-fame   
2  01-03-2014          700000.00               /organization/-qounter   
3  30-01-2014         3406878.00  /organization/-the-one-of-them-inc-   
4  19-03-2008         2000000.00                /organization/0-6-com   
6  01-07-2014           41250.00     /organization/01games-technology   

                     name                                      category_list  \
0                   #fame                                              Media   
2                :qounter  Application Platforms|Real Time|Social Network...   
3  (the) one of them,inc.                                  Apps|Games|Mobile   
4                 0-6.com                                        Curated Web   
6      01games technology                                              Games   

      status country_code  
0  operating          IND  
2  operating          USA  
3  operating          NaN  
4  operating          CHN  
6  operating          HKG

In [71]:
round(100*master_frame.isnull().sum()/len(master_frame.index),2)

funding_round_permalink   0.00
funding_round_type        0.00
funded_at                 0.00
raised_amount_usd         0.00
permalink                 0.00
name                      0.00
category_list             1.10
status                    0.00
country_code              6.16
dtype: float64

In [72]:
master_frame.isnull().sum()

funding_round_permalink       0
funding_round_type            0
funded_at                     0
raised_amount_usd             0
permalink                     0
name                          1
category_list              1044
status                        0
country_code               5850
dtype: int64

In [73]:
master_frame.shape

(94958, 9)

In [100]:
master_frame.to_csv('master.csv')

In [101]:
index_countries = np.where(master_frame['country_code'].isna())

In [102]:
print(index_countries, len(index_countries[0]))

(array([    2,    26,    52, ..., 94862, 94874, 94957], dtype=int64),) 5850


In [103]:
index_cat = np.where(master_frame['category_list'].isna())

In [104]:
print(index_cat, len(index_cat[0]))

(array([   75,   102,   141, ..., 94443, 94539, 94540], dtype=int64),) 1044


In [90]:
bool_cat = pd.isnull(master_frame['category_list'])
print(len(index_cat))

94958


In [107]:
# print(bool_cat)

In [86]:
# categories can have Others as primary_sector; so lets fill the category data

In [106]:
master_frame.loc[bool_cat, ['category_list']] = "unknown_cat"

In [108]:
master_frame.isnull().sum()

funding_round_permalink       0
funding_round_type            0
funded_at                     0
raised_amount_usd             0
permalink                     0
name                          1
category_list                 0
status                        0
country_code               5850
dtype: int64

In [109]:
bool_country = pd.isnull(master_frame['country_code'])

In [110]:
master_frame.loc[bool_country, ['country_code']] = "unknown_country"

In [112]:
round(100*master_frame.isnull().sum()/len(master_frame.index),2)

funding_round_permalink   0.00
funding_round_type        0.00
funded_at                 0.00
raised_amount_usd         0.00
permalink                 0.00
name                      0.00
category_list             0.00
status                    0.00
country_code              0.00
dtype: float64

#### MAster frame is complete.

In [113]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 94958 entries, 0 to 114941
Data columns (total 9 columns):
funding_round_permalink    94958 non-null object
funding_round_type         94958 non-null object
funded_at                  94958 non-null object
raised_amount_usd          94958 non-null float64
permalink                  94958 non-null object
name                       94957 non-null object
category_list              94958 non-null object
status                     94958 non-null object
country_code               94958 non-null object
dtypes: float64(1), object(8)
memory usage: 9.7+ MB


In [114]:
master_frame.describe()

raised_amount_usd
count           94958.00
mean         10426962.02
std         114821849.03
min                 0.00
25%            322500.00
50%           1680521.00
75%           7000000.00
max       21271935000.00

### CheckPoint2

In [133]:
f_mean = master_frame.groupby('funding_round_type').mean()
print(type(f_mean))
f_mean.sort_values(ascending=True, by=['raised_amount_usd'])

<class 'pandas.core.frame.DataFrame'>


raised_amount_usd
funding_round_type                      
non_equity_assistance          411203.05
equity_crowdfunding            538368.21
seed                           719779.62
angel                          958694.47
product_crowdfunding          1363131.07
convertible_note              1453438.54
grant                         4300576.34
venture                      11748949.13
debt_financing               17043526.02
undisclosed                  19242370.23
private_equity               73308593.03
secondary_market             79649630.10
post_ipo_equity              82182493.87
post_ipo_debt               168704571.82

In [134]:
f_mean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14 entries, angel to venture
Data columns (total 1 columns):
raised_amount_usd    14 non-null float64
dtypes: float64(1)
memory usage: 224.0+ bytes


In [135]:
f_mean.columns

Index(['raised_amount_usd'], dtype='object')

In [136]:
print(master_frame.funding_round_type.unique())

['venture' 'seed' 'undisclosed' 'convertible_note' 'private_equity'
 'debt_financing' 'angel' 'grant' 'equity_crowdfunding' 'post_ipo_equity'
 'post_ipo_debt' 'product_crowdfunding' 'secondary_market'
 'non_equity_assistance']


In [138]:
f_venture = master_frame['funding_round_type']=='venture'
f_seed = master_frame['funding_round_type']=='seed'
f_pequity = master_frame['funding_round_type']=='private_equity'
f_angel = master_frame['funding_round_type']=='angel'

In [139]:
df_venture = master_frame[f_venture]
df_seed = master_frame[f_seed]
df_pequity = master_frame[f_pequity]
df_angel = master_frame[f_angel]

### Answers to the Question
Average funding amount of venture type

Average funding amount of angel type

Average funding amount of seed type

Average funding amount of private equity type


In [140]:
print('Venture - ' + str(round(df_venture.raised_amount_usd.mean(),2)))
print('Seed - ' + str(round(df_seed.raised_amount_usd.mean(),2)))
print('Private Equity - ' + str(round(df_pequity.raised_amount_usd.mean(),2)))
print('Angel - ' + str(round(df_angel.raised_amount_usd.mean(),2)))

Venture - 11748949.13
Seed - 719779.62
Private Equity - 73308593.03
Angel - 958694.47


In [141]:
m_venture = df_venture.raised_amount_usd.mean()
m_seed = df_seed.raised_amount_usd.mean()
m_pequity = df_pequity.raised_amount_usd.mean()
m_angel = df_angel.raised_amount_usd.mean()

print(m_venture, m_seed, m_pequity, m_angel)

11748949.129489528 719779.6202016778 73308593.02944215 958694.4697530865


In [142]:
# 5 to 15 million
lower = 5000000
upper = 15000000

In [143]:
m_venture>=lower and m_venture<upper

True

In [144]:
m_seed>=lower and m_seed<=upper

False

In [145]:
m_pequity>=lower and m_pequity<=upper

False

In [146]:
m_angel>=lower and m_angel<=upper

False

### [Answer] Q. Considering that Spark Funds wants to invest between 5 to 15 million USD per  investment round, which investment type is the most suitable for them?


### Checkpoint 3: Country Analysis

In [148]:
# From the previous analysis - venture fund is suitable to invest

In [149]:
df_venture.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 50228 entries, 0 to 114934
Data columns (total 9 columns):
funding_round_permalink    50228 non-null object
funding_round_type         50228 non-null object
funded_at                  50228 non-null object
raised_amount_usd          50228 non-null float64
permalink                  50228 non-null object
name                       50228 non-null object
category_list              50228 non-null object
status                     50228 non-null object
country_code               50228 non-null object
dtypes: float64(1), object(8)
memory usage: 3.8+ MB


In [150]:
df_venture.describe()

raised_amount_usd
count           50228.00
mean         11748949.13
std          86352066.56
min                 0.00
25%           1600901.75
50%           5000000.00
75%          12000000.00
max       17600000000.00

In [151]:
df_venture.head(10)

funding_round_permalink funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
3   /funding-round/650b8f704416801069bb178a1418776b            venture   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
12  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
22  /funding-round/b952cbaf401f310927430c97b68162ea            venture   
28  /funding-round/0faccbbcc5818dc5326469f13f5a8ac8            venture   
34  /funding-round/502bd0e50c27616995e4bdad24605ef8            venture   

     funded_at  raised_amount_usd                            permalink  \
0   05-01-2015        10000000.00                  /organization/-fame   
3   30-01-2014         3406878.00  /organization/-the-one-of-them-inc-   
4   19-03-2008         2000000.00                /organization/0-6-com   
8   21-12-2009          719491.00  /organization/0ndine-biomedical-inc   
10  09-11-2015        20000000.00                 /organization/0xdata   
11  03-01-2013         1700000.00                 /organization/0xdata   
12  19-07-2014         8900000.00                 /organization/0xdata   
22  17-03-2015         5000000.00           /organization/1-mainstream   
28  09-10-2014         4000000.00        /organization/10-minutes-with   
34  16-02-2011         2520000.00           /organization/1000memories   

                      name                     category_list     status  \
0                    #fame                             Media  operating   
3   (the) one of them,inc.                 Apps|Games|Mobile  operating   
4                  0-6.com                       Curated Web  operating   
8   ondine biomedical inc.                     Biotechnology  operating   
10                  h2o.ai                         Analytics  operating   
11                  h2o.ai                         Analytics  operating   
12                  h2o.ai                         Analytics  operating   
22            1 mainstream  Apps|Cable|Distribution|Software   acquired   
28         10 minutes with                         Education  operating   
34            1000memories                       Curated Web   acquired   

       country_code  
0               IND  
3   unknown_country  
4               CHN  
8               CAN  
10              USA  
11              USA  
12              USA  
22              USA  
28              GBR  
34              USA

In [152]:
df_venture['category_list'] = df_venture['category_list'].astype(str)

C:\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [153]:
# df_venture =df_venture[pd.notnull(df_venture['country_code'])]

In [154]:
df_venture.country_code.value_counts().nlargest(9)

USA                36139
unknown_country     2117
GBR                 2055
CHN                 1564
CAN                 1256
FRA                  917
IND                  824
ISR                  796
DEU                  558
Name: country_code, dtype: int64

In [156]:
df_venture.columns

Index(['funding_round_permalink', 'funding_round_type', 'funded_at',
       'raised_amount_usd', 'permalink', 'name', 'category_list', 'status',
       'country_code'],
      dtype='object')

### 1. Spark Funds wants to see the top nine countries which have received the highest total funding (across ALL sectors for the chosen investment type)

In [168]:
df_venture.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False).head(15)

country_code
USA               422510842796.00
CHN                39835418773.00
unknown_country    25506108414.00
GBR                20245627416.00
IND                14391858718.00
CAN                 9583332317.00
FRA                 7259536732.00
ISR                 6907514579.00
DEU                 6346959822.00
JPN                 3363676611.00
SWE                 3254952563.00
NLD                 2939403619.00
CHE                 2827560264.00
SGP                 2793917856.00
ESP                 1835831452.00
Name: raised_amount_usd, dtype: float64

#### Ignoring the unknown country as we dont know what are those

In [167]:
top9_countries = ['USA','CHN', 'GBR', 'IND', 'CAN','FRA','ISR', 'DEU', 'JPN']

### 2. For the chosen investment type, make a data frame named top9 with the top nine countries (based on the total investment amount each country has received)

In [169]:
top9 = df_venture[df_venture['country_code'].isin(top9_countries)]

In [170]:
top9.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 44383 entries, 0 to 114934
Data columns (total 9 columns):
funding_round_permalink    44383 non-null object
funding_round_type         44383 non-null object
funded_at                  44383 non-null object
raised_amount_usd          44383 non-null float64
permalink                  44383 non-null object
name                       44383 non-null object
category_list              44383 non-null object
status                     44383 non-null object
country_code               44383 non-null object
dtypes: float64(1), object(8)
memory usage: 3.4+ MB


In [171]:
top9.head(15)

funding_round_permalink funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
12  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
22  /funding-round/b952cbaf401f310927430c97b68162ea            venture   
28  /funding-round/0faccbbcc5818dc5326469f13f5a8ac8            venture   
34  /funding-round/502bd0e50c27616995e4bdad24605ef8            venture   
38  /funding-round/13be128d655076a025221d7fddc90d68            venture   
39  /funding-round/6aeb32ab3fc05db2b606d7a2466e8c09            venture   
41  /funding-round/89317984885b44f02e4befb9dc2e588c            venture   
44  /funding-round/fd1b3064232ba9b3cfb5653a619e82e8            venture   
46  /funding-round/645b4cd895450b8e0268027cd7813047            venture   
47  /funding-round/8025123b90c1ab687f9857ffe18bfd3b            venture   

     funded_at  raised_amount_usd                            permalink  \
0   05-01-2015        10000000.00                  /organization/-fame   
4   19-03-2008         2000000.00                /organization/0-6-com   
8   21-12-2009          719491.00  /organization/0ndine-biomedical-inc   
10  09-11-2015        20000000.00                 /organization/0xdata   
11  03-01-2013         1700000.00                 /organization/0xdata   
12  19-07-2014         8900000.00                 /organization/0xdata   
22  17-03-2015         5000000.00           /organization/1-mainstream   
28  09-10-2014         4000000.00        /organization/10-minutes-with   
34  16-02-2011         2520000.00           /organization/1000memories   
38  14-10-2009          265940.00        /organization/1000museums-com   
39  15-01-2014         2906832.00        /organization/1000museums-com   
41  01-08-2014         1500000.00        /organization/1000museums-com   
44  05-04-2011          642000.00        /organization/1000museums-com   
46  04-11-2015         6602694.00             /organization/1001-menus   
47  13-11-2013         1736910.00             /organization/1001-menus   

                      name                     category_list     status  \
0                    #fame                             Media  operating   
4                  0-6.com                       Curated Web  operating   
8   ondine biomedical inc.                     Biotechnology  operating   
10                  h2o.ai                         Analytics  operating   
11                  h2o.ai                         Analytics  operating   
12                  h2o.ai                         Analytics  operating   
22            1 mainstream  Apps|Cable|Distribution|Software   acquired   
28         10 minutes with                         Education  operating   
34            1000memories                       Curated Web   acquired   
38         1000museums.com                       Curated Web  operating   
39         1000museums.com                       Curated Web  operating   
41         1000museums.com                       Curated Web  operating   
44         1000museums.com                       Curated Web  operating   
46                 zenchef      Local Businesses|Restaurants  operating   
47                 zenchef      Local Businesses|Restaurants  operating   

   country_code  
0           IND  
4           CHN  
8           CAN  
10          USA  
11          USA  
12          USA  
22          USA  
28          GBR  
34          USA  
38          USA  
39          USA  
41          USA  
44          USA  
46          FRA  
47          FRA

In [172]:
top9.describe()

raised_amount_usd
count           44383.00
mean         11951530.27
std          91291415.39
min                 0.00
25%           1700000.00
50%           5000000.00
75%          12000000.00
max       17600000000.00

### Identify the top three English-speaking countries in the data frame top9.

In [173]:
# https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes
list_eng_countries = ['GBR', 'AUS', 'BHS', 'BWA', 'CAN','CYP', 'FJI', 'IND', 
                      'KEN', 'KIR', 'MLT', 'NGA', 'PAK', 'PNG', 'IRL', 'ZAF', 
                      'NZL', 'SGP', 'PHL', 'GMB']
print(len(list_eng_countries))

20


In [174]:
countries = 'ATG,AUS,BHS,BRB,BLZ,BWA,CMR,CAN,DMA,ERI,ETH,FJI,GMB,GHA,GRD,GUY,IND,IRL,JAM,KEN,KIR,NLD,LSO,LBR,MWI,MLT,MHL,MUS,FSM,NAM,NRU,NZL,NGA,PAK,PLW,PNG,PHL,RWA,KNA,LCA,VCT,WSM,SYC,SLE,SGP,SLB,ZAF,SSD,SDN,SWZ,TZA,TON,TTO,TUV,GBR,USA,UGA,VUT,ZMB,ZWE'
eng_countries = countries.split(',')
print(len(eng_countries))

60


In [175]:
list_eng_countries = list(set(list_eng_countries)|set(eng_countries))
print(type(list_eng_countries), len(list_eng_countries), list_eng_countries)

<class 'list'> 61 ['CMR', 'ZWE', 'BWA', 'TZA', 'DMA', 'IRL', 'GMB', 'FSM', 'FJI', 'CAN', 'MWI', 'MHL', 'LCA', 'BHS', 'NGA', 'TTO', 'KNA', 'JAM', 'GBR', 'SLE', 'WSM', 'SWZ', 'TON', 'PLW', 'MLT', 'ZMB', 'BLZ', 'SDN', 'VUT', 'GRD', 'KIR', 'LBR', 'NLD', 'PAK', 'AUS', 'LSO', 'SGP', 'GUY', 'UGA', 'ETH', 'USA', 'ERI', 'ZAF', 'PNG', 'RWA', 'CYP', 'GHA', 'PHL', 'SSD', 'VCT', 'KEN', 'MUS', 'NAM', 'NZL', 'SLB', 'IND', 'SYC', 'NRU', 'ATG', 'TUV', 'BRB']


In [176]:
list_eng_countries.sort()

In [177]:
print(list_eng_countries, len(list_eng_countries))

['ATG', 'AUS', 'BHS', 'BLZ', 'BRB', 'BWA', 'CAN', 'CMR', 'CYP', 'DMA', 'ERI', 'ETH', 'FJI', 'FSM', 'GBR', 'GHA', 'GMB', 'GRD', 'GUY', 'IND', 'IRL', 'JAM', 'KEN', 'KIR', 'KNA', 'LBR', 'LCA', 'LSO', 'MHL', 'MLT', 'MUS', 'MWI', 'NAM', 'NGA', 'NLD', 'NRU', 'NZL', 'PAK', 'PHL', 'PLW', 'PNG', 'RWA', 'SDN', 'SGP', 'SLB', 'SLE', 'SSD', 'SWZ', 'SYC', 'TON', 'TTO', 'TUV', 'TZA', 'UGA', 'USA', 'VCT', 'VUT', 'WSM', 'ZAF', 'ZMB', 'ZWE'] 61


In [178]:
top9_eng = top9[top9['country_code'].isin(list_eng_countries)]

In [180]:
top9_eng.country_code.value_counts()

USA    36139
GBR     2055
CAN     1256
IND      824
Name: country_code, dtype: int64

### Top3 english speaking countries - Investment wise

#### 1.USA
#### 2. GBR
#### 3. CAN

In [246]:
top9_eng.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 40274 entries, 0 to 114929
Data columns (total 9 columns):
funding_round_permalink    40274 non-null object
funding_round_type         40274 non-null object
funded_at                  40274 non-null object
raised_amount_usd          40274 non-null float64
permalink                  40274 non-null object
name                       40274 non-null object
category_list              40274 non-null object
status                     40274 non-null object
country_code               40274 non-null object
dtypes: float64(1), object(8)
memory usage: 3.1+ MB


In [249]:
top3_country_list = ['USA','GBR','CAN']

In [251]:
top3 = top9_eng[top9_eng['country_code'].isin(top3_country_list)]

In [252]:
top3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 39450 entries, 8 to 114929
Data columns (total 9 columns):
funding_round_permalink    39450 non-null object
funding_round_type         39450 non-null object
funded_at                  39450 non-null object
raised_amount_usd          39450 non-null float64
permalink                  39450 non-null object
name                       39450 non-null object
category_list              39450 non-null object
status                     39450 non-null object
country_code               39450 non-null object
dtypes: float64(1), object(8)
memory usage: 3.0+ MB


### Checkpoint 4: Sector Analysis 1

In [181]:
df_mapping = pd.read_csv('mapping.csv',encoding = "ISO-8859-1")

In [182]:
df_mapping.columns

Index(['category_list', 'Automotive & Sports', 'Blanks',
       'Cleantech / Semiconductors', 'Entertainment', 'Health',
       'Manufacturing', 'News, Search and Messaging', 'Others',
       'Social, Finance, Analytics, Advertising'],
      dtype='object')

In [183]:
df_mapping.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 10 columns):
category_list                              687 non-null object
Automotive & Sports                        688 non-null int64
Blanks                                     688 non-null int64
Cleantech / Semiconductors                 688 non-null int64
Entertainment                              688 non-null int64
Health                                     688 non-null int64
Manufacturing                              688 non-null int64
News, Search and Messaging                 688 non-null int64
Others                                     688 non-null int64
Social, Finance, Analytics, Advertising    688 non-null int64
dtypes: int64(9), object(1)
memory usage: 53.8+ KB


In [184]:
df_mapping.head(10)

category_list  Automotive & Sports  Blanks  \
0                 NaN                    0       1   
1                  3D                    0       0   
2         3D Printing                    0       0   
3       3D Technology                    0       0   
4          Accounting                    0       0   
5    Active Lifestyle                    0       0   
6        Ad Targeting                    0       0   
7  Advanced Materials                    0       0   
8    Adventure Travel                    1       0   
9         Advertising                    0       0   

   Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
0                           0              0       0              0   
1                           0              0       0              1   
2                           0              0       0              1   
3                           0              0       0              1   
4                           0              0       0              0   
5                           0              0       1              0   
6                           0              0       0              0   
7                           0              0       0              1   
8                           0              0       0              0   
9                           0              0       0              0   

   News, Search and Messaging  Others  Social, Finance, Analytics, Advertising  
0                           0       0                                        0  
1                           0       0                                        0  
2                           0       0                                        0  
3                           0       0                                        0  
4                           0       0                                        1  
5                           0       0                                        0  
6                           0       0                                        1  
7                           0       0                                        0  
8                           0       0                                        0  
9                           0       0                                        1

In [186]:
def reCategorize(val):
    splits = val.split('|')
    if(len(splits)>1):
        return splits[0]
    else:
        return val

In [187]:
print(reCategorize('Apps|Cable|Distribution|Software'))

Apps


In [188]:
print(reCategorize('Analytics'))

Analytics


In [189]:
print(reCategorize('Curated Web'))

Curated Web


In [192]:
# df_venture['primary_sector'] = df_venture['category_list'].apply(reCategorize)

C:\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [193]:
master_frame['primary_sector'] = master_frame['category_list'].apply(reCategorize)

In [194]:
master_frame.head()

funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   

    funded_at  raised_amount_usd                            permalink  \
0  05-01-2015        10000000.00                  /organization/-fame   
2  01-03-2014          700000.00               /organization/-qounter   
3  30-01-2014         3406878.00  /organization/-the-one-of-them-inc-   
4  19-03-2008         2000000.00                /organization/0-6-com   
6  01-07-2014           41250.00     /organization/01games-technology   

                     name                                      category_list  \
0                   #fame                                              Media   
2                :qounter  Application Platforms|Real Time|Social Network...   
3  (the) one of them,inc.                                  Apps|Games|Mobile   
4                 0-6.com                                        Curated Web   
6      01games technology                                              Games   

      status     country_code         primary_sector  
0  operating              IND                  Media  
2  operating              USA  Application Platforms  
3  operating  unknown_country                   Apps  
4  operating              CHN            Curated Web  
6  operating              HKG                  Games

In [195]:
master_frame['primary_sector'] = master_frame['primary_sector'].str.lower()

In [196]:
master_frame.head()

funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   

    funded_at  raised_amount_usd                            permalink  \
0  05-01-2015        10000000.00                  /organization/-fame   
2  01-03-2014          700000.00               /organization/-qounter   
3  30-01-2014         3406878.00  /organization/-the-one-of-them-inc-   
4  19-03-2008         2000000.00                /organization/0-6-com   
6  01-07-2014           41250.00     /organization/01games-technology   

                     name                                      category_list  \
0                   #fame                                              Media   
2                :qounter  Application Platforms|Real Time|Social Network...   
3  (the) one of them,inc.                                  Apps|Games|Mobile   
4                 0-6.com                                        Curated Web   
6      01games technology                                              Games   

      status     country_code         primary_sector  
0  operating              IND                  media  
2  operating              USA  application platforms  
3  operating  unknown_country                   apps  
4  operating              CHN            curated web  
6  operating              HKG                  games

In [ ]:
# primary sector column has been created with first category but not mapped according the mapping.csv; lets do that.

In [197]:
df_mapping.shape

(688, 10)

In [198]:
df_mapping.head(10)

category_list  Automotive & Sports  Blanks  \
0                 NaN                    0       1   
1                  3D                    0       0   
2         3D Printing                    0       0   
3       3D Technology                    0       0   
4          Accounting                    0       0   
5    Active Lifestyle                    0       0   
6        Ad Targeting                    0       0   
7  Advanced Materials                    0       0   
8    Adventure Travel                    1       0   
9         Advertising                    0       0   

   Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
0                           0              0       0              0   
1                           0              0       0              1   
2                           0              0       0              1   
3                           0              0       0              1   
4                           0              0       0              0   
5                           0              0       1              0   
6                           0              0       0              0   
7                           0              0       0              1   
8                           0              0       0              0   
9                           0              0       0              0   

   News, Search and Messaging  Others  Social, Finance, Analytics, Advertising  
0                           0       0                                        0  
1                           0       0                                        0  
2                           0       0                                        0  
3                           0       0                                        0  
4                           0       0                                        1  
5                           0       0                                        0  
6                           0       0                                        1  
7                           0       0                                        0  
8                           0       0                                        0  
9                           0       0                                        1

In [199]:
df_mapping_melt = pd.melt(df_mapping, id_vars=['category_list'])

In [200]:
df_mapping_melt.shape

(6192, 3)

In [201]:
type(df_mapping_melt)

pandas.core.frame.DataFrame

In [202]:
df_mapping_melt = df_mapping_melt[df_mapping_melt['value']!=0]

In [203]:
df_mapping_melt.shape

(688, 3)

In [205]:
df_mapping_melt.head()

category_list             variable  value
8   Adventure Travel  Automotive & Sports      1
14         Aerospace  Automotive & Sports      1
45              Auto  Automotive & Sports      1
46   Automated Kiosk  Automotive & Sports      1
47        Automotive  Automotive & Sports      1

In [204]:
df_mapping_melt.columns

Index(['category_list', 'variable', 'value'], dtype='object')

In [207]:
df_mapping_melt.drop('value', axis=1, inplace=True)

In [208]:
df_mapping_melt.columns

Index(['category_list', 'variable'], dtype='object')

In [210]:
df_mapping_melt['category_list'] = df_mapping_melt['category_list'].str.lower()

In [215]:
df_mapping_melt.rename(columns={'category_list':'primary_sector'}, inplace=True)

In [220]:
df_mapping_melt.to_csv('mapping_melt.csv')

There are some spelling mistakes in the mapping file on observation; there are 0 present in case where na should be present. so applying regex and correct them

In [221]:
df_mapping_melt['primary_sector'] = df_mapping_melt.primary_sector.apply(lambda x: re.sub('[0]', 'na', str(x)))

In [222]:
df_mapping_melt.to_csv('mapping_melt_corrected.csv')

In [228]:
# master_frame.drop(columns=['_merge','variable'], inplace=True)

If we dont rename, it creating two columns with category_list_x and category_list_y; so rename and merge

In [226]:
master_frame = pd.merge(master_frame, df_mapping_melt, how='left', on='primary_sector',indicator=True)


In [229]:
master_frame.head()

funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
2  /funding-round/650b8f704416801069bb178a1418776b            venture   
3  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
4  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   

    funded_at  raised_amount_usd                            permalink  \
0  05-01-2015        10000000.00                  /organization/-fame   
1  01-03-2014          700000.00               /organization/-qounter   
2  30-01-2014         3406878.00  /organization/-the-one-of-them-inc-   
3  19-03-2008         2000000.00                /organization/0-6-com   
4  01-07-2014           41250.00     /organization/01games-technology   

                     name                                      category_list  \
0                   #fame                                              Media   
1                :qounter  Application Platforms|Real Time|Social Network...   
2  (the) one of them,inc.                                  Apps|Games|Mobile   
3                 0-6.com                                        Curated Web   
4      01games technology                                              Games   

      status     country_code         primary_sector  \
0  operating              IND                  media   
1  operating              USA  application platforms   
2  operating  unknown_country                   apps   
3  operating              CHN            curated web   
4  operating              HKG                  games   

                     variable _merge  
0               Entertainment   both  
1  News, Search and Messaging   both  
2  News, Search and Messaging   both  
3  News, Search and Messaging   both  
4               Entertainment   both

In [230]:
round(100*master_frame.isnull().sum()/len(master_frame.index),2)

funding_round_permalink   0.00
funding_round_type        0.00
funded_at                 0.00
raised_amount_usd         0.00
permalink                 0.00
name                      0.00
category_list             0.00
status                    0.00
country_code              0.00
primary_sector            0.00
variable                  1.19
_merge                    0.00
dtype: float64

In [231]:
master_frame.variable.unique()

array(['Entertainment', 'News, Search and Messaging',
       'Cleantech / Semiconductors',
       'Social, Finance, Analytics, Advertising', 'Others', 'Health', nan,
       'Manufacturing', 'Automotive & Sports'], dtype=object)

In [234]:
master_frame.rename(columns={'variable':'main_sector'}, inplace=True)

In [235]:
master_frame.to_csv('master_1.csv')

## Checkpoint 5: Sector Analysis 2

In [238]:
master_frame.columns

Index(['funding_round_permalink', 'funding_round_type', 'funded_at',
       'raised_amount_usd', 'permalink', 'name', 'category_list', 'status',
       'country_code', 'primary_sector', 'main_sector', '_merge'],
      dtype='object')

In [240]:
f_venture = master_frame['funding_round_type']=='venture'
df_venture = master_frame[f_venture]

In [253]:
df_top3 = df_venture[df_venture['country_code'].isin(top3_country_list)]

In [254]:
df_top3.head()

funding_round_permalink funding_round_type  \
6   /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
8   /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
9   /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
10  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
16  /funding-round/b952cbaf401f310927430c97b68162ea            venture   

     funded_at  raised_amount_usd                            permalink  \
6   21-12-2009          719491.00  /organization/0ndine-biomedical-inc   
8   09-11-2015        20000000.00                 /organization/0xdata   
9   03-01-2013         1700000.00                 /organization/0xdata   
10  19-07-2014         8900000.00                 /organization/0xdata   
16  17-03-2015         5000000.00           /organization/1-mainstream   

                      name                     category_list     status  \
6   ondine biomedical inc.                     Biotechnology  operating   
8                   h2o.ai                         Analytics  operating   
9                   h2o.ai                         Analytics  operating   
10                  h2o.ai                         Analytics  operating   
16            1 mainstream  Apps|Cable|Distribution|Software   acquired   

   country_code primary_sector                              main_sector _merge  
6           CAN  biotechnology               Cleantech / Semiconductors   both  
8           USA      analytics  Social, Finance, Analytics, Advertising   both  
9           USA      analytics  Social, Finance, Analytics, Advertising   both  
10          USA      analytics  Social, Finance, Analytics, Advertising   both  
16          USA           apps               News, Search and Messaging   both

In [255]:
df_top3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 39450 entries, 6 to 94946
Data columns (total 12 columns):
funding_round_permalink    39450 non-null object
funding_round_type         39450 non-null object
funded_at                  39450 non-null object
raised_amount_usd          39450 non-null float64
permalink                  39450 non-null object
name                       39450 non-null object
category_list              39450 non-null object
status                     39450 non-null object
country_code               39450 non-null object
primary_sector             39450 non-null object
main_sector                39215 non-null object
_merge                     39450 non-null category
dtypes: category(1), float64(1), object(10)
memory usage: 3.6+ MB


In [256]:
df_top3.shape

(39450, 12)

In [257]:
df_top3.country_code.value_counts()

USA    36139
GBR     2055
CAN     1256
Name: country_code, dtype: int64

In [258]:
df_top3.to_csv('top3.csv')

In [259]:
val2 = 20000000.00
val2>=lower and val2<upper

False

If we observe there are some investment which are not falling in between 5 and 15 millioin, lets filter out

In [263]:
# df_top3.drop(df_top3[(df_top3.raised_amount_usd < 5000000)].index, inplace=True)

In [264]:
 df_top3 = df_top3.drop(df_top3[(df_top3.raised_amount_usd < 5000000)].index)

In [265]:
 df_top3 = df_top3.drop(df_top3[(df_top3.raised_amount_usd > 15000000)].index)

In [266]:
df_top3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13204 entries, 10 to 94946
Data columns (total 12 columns):
funding_round_permalink    13204 non-null object
funding_round_type         13204 non-null object
funded_at                  13204 non-null object
raised_amount_usd          13204 non-null float64
permalink                  13204 non-null object
name                       13204 non-null object
category_list              13204 non-null object
status                     13204 non-null object
country_code               13204 non-null object
primary_sector             13204 non-null object
main_sector                13106 non-null object
_merge                     13204 non-null category
dtypes: category(1), float64(1), object(10)
memory usage: 1.2+ MB


In [268]:
df_top3.dropna(subset=['main_sector'], inplace=True)

In [269]:
df_top3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13106 entries, 10 to 94946
Data columns (total 12 columns):
funding_round_permalink    13106 non-null object
funding_round_type         13106 non-null object
funded_at                  13106 non-null object
raised_amount_usd          13106 non-null float64
permalink                  13106 non-null object
name                       13106 non-null object
category_list              13106 non-null object
status                     13106 non-null object
country_code               13106 non-null object
primary_sector             13106 non-null object
main_sector                13106 non-null object
_merge                     13106 non-null category
dtypes: category(1), float64(1), object(10)
memory usage: 1.2+ MB


In [271]:
df_top3.drop('_merge', axis=1, inplace=True)

In [272]:
df_top3.head()

funding_round_permalink funding_round_type  \
10  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
16  /funding-round/b952cbaf401f310927430c97b68162ea            venture   
78  /funding-round/fb6216a30cb566ede89e0bee0623a634            venture   
84  /funding-round/424129ce1235cfab2655ee81305f7c2b            venture   
85  /funding-round/6d3f3797371956ece035b8478c1441b2            venture   

     funded_at  raised_amount_usd                        permalink  \
10  19-07-2014         8900000.00             /organization/0xdata   
16  17-03-2015         5000000.00       /organization/1-mainstream   
78  16-12-2014        11999347.00     /organization/128-technology   
84  15-10-2013        15000000.00  /organization/1366-technologies   
85  09-04-2015         5000000.00  /organization/1366-technologies   

                 name                     category_list     status  \
10             h2o.ai                         Analytics  operating   
16       1 mainstream  Apps|Cable|Distribution|Software   acquired   
78     128 technology      Service Providers|Technology  operating   
84  1366 technologies                     Manufacturing  operating   
85  1366 technologies                     Manufacturing  operating   

   country_code     primary_sector                              main_sector  
10          USA          analytics  Social, Finance, Analytics, Advertising  
16          USA               apps               News, Search and Messaging  
78          USA  service providers                                   Others  
84          USA      manufacturing                            Manufacturing  
85          USA      manufacturing                            Manufacturing

#### D1, D2, D3 Dataframe
D1 - USA

D2 - GBR

D3 - CAN

In [273]:
d1 = df_top3[df_top3['country_code'] == 'USA']
d2 = df_top3[df_top3['country_code'] == 'GBR']
d3 = df_top3[df_top3['country_code'] == 'CAN']

In [274]:
print(d1.shape)
print(d2.shape)
print(d3.shape)

(12063, 11)
(621, 11)
(422, 11)


In [275]:
# The total number (or count) of investments for each main sector in a separate column

In [276]:
d1.columns

Index(['funding_round_permalink', 'funding_round_type', 'funded_at',
       'raised_amount_usd', 'permalink', 'name', 'category_list', 'status',
       'country_code', 'primary_sector', 'main_sector'],
      dtype='object')

### D1

In [283]:
d1.raised_amount_usd.sum()

107757097294.0

In [279]:
d1.pivot_table(values='raised_amount_usd', index='main_sector',aggfunc=['sum',np.size])

sum              size
                                        raised_amount_usd raised_amount_usd
main_sector                                                                
Automotive & Sports                         1454104361.00            167.00
Cleantech / Semiconductors                 21633430822.00           2350.00
Entertainment                               5099197982.00            591.00
Health                                      8211859357.00            909.00
Manufacturing                               7258553378.00            799.00
News, Search and Messaging                 13971567428.00           1583.00
Others                                     26321007002.00           2950.00
Social, Finance, Analytics, Advertising    23807376964.00           2714.00

In [286]:
f_d1_others = d1['main_sector']=='Others'

In [287]:
d1_others = d1[f_d1_others]

In [288]:
d1_others.head()

funding_round_permalink funding_round_type  \
78   /funding-round/fb6216a30cb566ede89e0bee0623a634            venture   
103  /funding-round/b84bb882ca873f5fb96535671981196d            venture   
109  /funding-round/69690484f51e15bc27ff52bfe472cd96            venture   
110  /funding-round/8d87f771e938e0f31641bd600abbafca            venture   
147  /funding-round/292b074d073fdd9c7e9d8f372c3aa5f6            venture   

      funded_at  raised_amount_usd                     permalink  \
78   16-12-2014        11999347.00  /organization/128-technology   
103  16-04-2002        14000000.00     /organization/170-systems   
109  01-01-2011         5000000.00         /organization/17zuoye   
110  01-09-2013        10000000.00         /organization/17zuoye   
147  24-01-2014        15000000.00         /organization/1stdibs   

               name                 category_list     status country_code  \
78   128 technology  Service Providers|Technology  operating          USA   
103     170 systems                      Software   acquired          USA   
109         17zuoye   Education|Language Learning  operating          USA   
110         17zuoye   Education|Language Learning  operating          USA   
147         1stdibs                    E-Commerce  operating          USA   

        primary_sector main_sector  
78   service providers      Others  
103           software      Others  
109          education      Others  
110          education      Others  
147         e-commerce      Others

In [292]:
d1_others.groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head()

permalink
/organization/virtustream           64300000.00
/organization/capella               54968051.00
/organization/airtight-networks     54201907.00
/organization/decarta               52100000.00
/organization/black-duck-software   51000000.00
Name: raised_amount_usd, dtype: float64

In [294]:
f_d1_sfaa = d1['main_sector']=='Social, Finance, Analytics, Advertising'
d1_sfaa = d1[f_d1_sfaa]
d1_sfaa.head()

funding_round_permalink funding_round_type  \
10   /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
180  /funding-round/bdf9f5bf67ee51155eae223acac57ec5            venture   
288  /funding-round/a45d977b48cb54216eb705d59b24ca19            venture   
290  /funding-round/bfc2233768b7c79ed58ad7561423e555            venture   
293  /funding-round/1c51042e815e96ed2653ae9ced99dfc4            venture   

      funded_at  raised_amount_usd                  permalink         name  \
10   19-07-2014         8900000.00       /organization/0xdata       h2o.ai   
180  17-11-2013         5050000.00         /organization/21e6       21 inc   
288  19-07-2012        13100000.00     /organization/33across     33across   
290  05-01-2011         9000000.00     /organization/33across     33across   
293  29-10-2003        12000000.00  /organization/360commerce  360commerce   

                                         category_list     status  \
10                                           Analytics  operating   
180    Big Data|Bitcoin|Hardware + Software|Technology  operating   
288  Advertising|Advertising Platforms|Content Disc...  operating   
290  Advertising|Advertising Platforms|Content Disc...  operating   
293             Information Technology|Retail|Software   acquired   

    country_code          primary_sector  \
10           USA               analytics   
180          USA                big data   
288          USA             advertising   
290          USA             advertising   
293          USA  information technology   

                                 main_sector  
10   Social, Finance, Analytics, Advertising  
180  Social, Finance, Analytics, Advertising  
288  Social, Finance, Analytics, Advertising  
290  Social, Finance, Analytics, Advertising  
293  Social, Finance, Analytics, Advertising

In [295]:
d1_sfaa.groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head()

permalink
/organization/shotspotter   67933006.00
/organization/demandbase    63000000.00
/organization/intacct       61800000.00
/organization/netbase       60600000.00
/organization/lotame        59700000.00
Name: raised_amount_usd, dtype: float64

### D2

In [284]:
d2.raised_amount_usd.sum()

5379078691.0

In [280]:
d2.pivot_table(values='raised_amount_usd', index='main_sector',aggfunc=['sum',np.size])

sum              size
                                        raised_amount_usd raised_amount_usd
main_sector                                                                
Automotive & Sports                          167051565.00             16.00
Cleantech / Semiconductors                  1163990056.00            130.00
Entertainment                                482784687.00             56.00
Health                                       214537510.00             24.00
Manufacturing                                361940335.00             42.00
News, Search and Messaging                   615746235.00             73.00
Others                                      1283624289.00            147.00
Social, Finance, Analytics, Advertising     1089404014.00            133.00

In [296]:
f_d2_others = d2['main_sector']=='Others'
d2_others = d2[f_d2_others]
d2_others.head()

funding_round_permalink funding_round_type  \
1462  /funding-round/064999587157b0ceae7843204a105d6d            venture   
1463  /funding-round/875b4295af2381558cec9d97f86fe3e4            venture   
2531  /funding-round/57ca5d05989850fb1109b277cb6ae54e            venture   
2532  /funding-round/ad79a10f4c3f231375454da6393ec564            venture   
3577  /funding-round/0db504d02c87c2b963283fd3c6d17594            venture   

       funded_at  raised_amount_usd                   permalink          name  \
1462  28-09-2015        13359377.00  /organization/acs-clothing  acs clothing   
1463  02-03-2014        12879637.00  /organization/acs-clothing  acs clothing   
2531  30-06-2010        10000000.00        /organization/aepona        aepona   
2532  05-02-2007        10000000.00        /organization/aepona        aepona   
3577  22-01-2008        13000000.00      /organization/alfresco      alfresco   

                                          category_list     status  \
1462                                         E-Commerce  operating   
1463                                         E-Commerce  operating   
2531                                        Web Hosting   acquired   
2532                                        Web Hosting   acquired   
3577  Document Management|Enterprises|Enterprise Sof...  operating   

     country_code       primary_sector main_sector  
1462          GBR           e-commerce      Others  
1463          GBR           e-commerce      Others  
2531          GBR          web hosting      Others  
2532          GBR          web hosting      Others  
3577          GBR  document management      Others

In [297]:
d2_others.groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head()

permalink
/organization/electric-cloud            37000000.00
/organization/sensage                   36250000.00
/organization/enigmatic                 32500000.00
/organization/silverrail-technologies   29000000.00
/organization/opencloud                 27972766.00
Name: raised_amount_usd, dtype: float64

In [299]:
f_d2_sfaa = d2['main_sector']=='Social, Finance, Analytics, Advertising'
d2_sfaa = d2[f_d2_sfaa]
d2_sfaa.head()
d2_sfaa.groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head()

permalink
/organization/celltick-technologies   37500000.00
/organization/mythings                34000000.00
/organization/zopa                    32900000.00
/organization/imagini                 28550000.00
/organization/marketinvoice           25553007.00
Name: raised_amount_usd, dtype: float64

### D3

In [285]:
d3.raised_amount_usd.sum()

3599289960.0

In [281]:
d3.pivot_table(values='raised_amount_usd', index='main_sector',aggfunc=['sum',np.size])

sum              size
                                        raised_amount_usd raised_amount_usd
main_sector                                                                
Automotive & Sports                           15000000.00              2.00
Cleantech / Semiconductors                  1015887607.00            112.00
Entertainment                                160022068.00             23.00
Health                                       168345064.00             21.00
Manufacturing                                237633071.00             31.00
News, Search and Messaging                   420121113.00             46.00
Others                                       926137962.00            109.00
Social, Finance, Analytics, Advertising      656143075.00             78.00

In [301]:
f_d3_cs = d3['main_sector']=='Cleantech / Semiconductors'
d3_cs = d3[f_d3_cs]
d3_cs.head()
d3_cs.groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head()

permalink
/organization/fresco-microchip      48000000.00
/organization/ostara                43152682.00
/organization/diablo-technologies   35500000.00
/organization/zymeworks             34100000.00
/organization/engene                31150000.00
Name: raised_amount_usd, dtype: float64

In [302]:
f_d3_others = d3['main_sector']=='Others'
d3_others = d3[f_d3_others]
d3_others.head()
d3_others.groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head()

permalink
/organization/quickplay-media        38700000.00
/organization/newstep                31477853.00
/organization/strangeloop-networks   21500000.00
/organization/morega                 20000000.00
/organization/lxdata                 19310000.00
Name: raised_amount_usd, dtype: float64

## Checkpoint 6: Plots